packages/protocol-kit/src/utils/erc-20/index.ts

```diff
@@ -0,0 +1,90 @@
+import { Interface } from '@ethersproject/abi'
+import Safe from '@safe-global/protocol-kit/Safe'
+import { Transaction } from '@safe-global/safe-core-sdk-types'
+
+import { ZERO_ADDRESS } from '../constants'
+
+const ERC20_ABI = [
+  'function transfer(address recipient, uint256 amount) returns (bool)',
+  'function decimals() view returns (uint8)'
+]
+
+/**
+ * Returns the number of decimals of a given ERC-20 token.
+ *
+ * @async
+ * @param {string} tokenAddress - The address of the ERC-20 token.
+ * @param {Safe} safe - The Safe object.
+ * @returns {Promise<number>} The number of decimals that the token uses.
+ */
+export async function getERC20Decimals(tokenAddress: string, safe: Safe): Promise<number> {
+  const ethAdapter = safe.getEthAdapter()
+  const erc20Interface = new Interface(ERC20_ABI)
+
+  const getTokenDecimalsTransaction = {
+    to: tokenAddress,
+    from: tokenAddress,
+    value: '0',
+    data: erc20Interface.encodeFunctionData('decimals')
+  }
+
+  const response = await ethAdapter.call(getTokenDecimalsTransaction)
+
+  const decimals = Number(response)
+
+  return decimals
+}
+
+const STANDARD_ERC20_DECIMALS = 18
+
+/**
+ * Checks if the given gas token is compatible with the handlePayment function present in the Safe smart contract.
+ * A token is considered compatible if it is a native token or a standard ERC-20 token with 18 decimals.
+ *
+ * @async
+ * @export
+ * @param {string} gasToken - The address of the gas token.
+ * @param {Safe} safe - The Safe object.
+ * @returns {Promise<boolean>} Returns true if the gas token is compatible, otherwise false.
+ */
+export async function isGasTokenCompatibleWithHandlePayment(
+  gasToken: string,
+  safe: Safe
+): Promise<boolean> {
+  const isNativeToken = gasToken === ZERO_ADDRESS
+
+  if (isNativeToken) {
+    return true
+  }
+
+  // Only ERC20 tokens with the standard 18 decimals are compatible
+  const gasTokenDecimals = await getERC20Decimals(gasToken, safe)
+  const isStandardERC20Token = gasTokenDecimals === STANDARD_ERC20_DECIMALS
```

Comment:
I would call this `hasTokenStandardERC20Decimals` because checking if it's a standard ERC20 would involve more things than just checking it has 18 decimals